synapse/storage/schema/main/delta/72/03thread_notifications.sql

```diff
@@ -0,0 +1,26 @@
+/* Copyright 2022 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ALTER TABLE event_push_actions_staging ADD COLUMN thread_id TEXT;
+
+ALTER TABLE event_push_actions ADD COLUMN thread_id TEXT;
```

Comment:
I wonder if the `thread_id` column should really live on the `events` table and not `event_push_actions` and `event_push_actions_staging`? This would be a bigger migration, but potentially more useful in the future? I don't have a strong opinion though.